src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -669,7 +669,6 @@ public OpenIdConnectHandler(HttpClient backchannel)
             var responseMessage = await Backchannel.SendAsync(requestMessage);
             responseMessage.EnsureSuccessStatusCode();
             var userInfoResponse = await responseMessage.Content.ReadAsStringAsync();
-            var userInfoEndpointJwt = new JwtSecurityToken(userInfoResponse);
             var user = JObject.Parse(userInfoResponse);
```

Comment:
@Tratcher I and @brentschmaltz talked about this and we were thinking if the protocol validator should take care of the claims merging so that the handler does not need to worry about the format of the response, which is what we wanted when we changed the type from JwtSecurityToken to string. On your end, you need to update the call to validate method to pass one extra parameter: ticket.Principal.Identity and also update the UserInfoReceivedContext to take 'userinfoendpointresponse' instead of 'user'. Validate method will return the identity with merged claims.